### DISCUSSION TITLE
Release v24.01.18 not published as Docker tagged image

### DISCUSSION THREAD

**Discussion by adamantike** [netalertx/NetAlertX#541] [General]
Latest release `24.01.18` has only be published to the Docker registry as `latest`. It seems the SemVer check didn't work correctly (), but specific tags should be available in the registry for users not planning on using `latest`.

Maybe the SemVer check from the GitHub action doesn't expect a leading zero in `01`?

**Comment by jokob-sk** [comment]
oh, thanks for the heads up. Do you think it's worth republishing the tag or you are OK to wait for the next release with a correct tag? I'm not sure how a setup using docker tags works so let me know what you'd prefer.

**Comment by adamantike** [comment]
If you plan to release soon, I think it's fine to wait for that new release!

I assume the error is related to the leading zero, because of the warning message [on this release action](https://github.com/jokob-sk/Pi.Alert/actions/runs/7566247368). According to SemVer specification, [item 2](https://semver.org/#spec-item-2):
> A normal version number MUST take the form X.Y.Z where X, Y, and Z are non-negative integers, and MUST NOT contain leading zeroes. X is the major version, Y is the minor version, and Z is the patch version. Each element MUST increase numerically. For instance: 1.9.0 -> 1.10.0 -> 1.11.0.

**Reply by jokob-sk** [reply]
Yes, it happened before - I release once a month so this will be fixed in 2-3 weeks. Thanks again.

**Comment by adamantike** [comment]
Closing this, as version `v24.2.17` has been correctly published to the registry! Thanks @jokob-sk

**Reply by jokob-sk** [reply]
This time, I used the correct convention 😅I was trying to figure out how to set up a check for the publishing to fail if the version doesn't represent a valid SemVer tag but couldn't find a quick fix. Let me know if you have ideas :)

**Reply by adamantike** [reply]
Currently, GitHub workflows are set to run on any tag matching `'*.*.*'`. That pattern could be changed to something like the following, to avoid being triggered when a leading zero is present: `'*.[1-9]+[0-9]?.[1-9]+*'`.

I haven't tested it, but it should avoid the minor/patch part to begin with zero, by not triggering the GitHub Action.

**Reply by jokob-sk** [reply]
Hey @adamantike - good idea, I will change it to make the process a bit less error-prone. Thanks for the suggestion!
